packages/invoice-template/src/common/LabeledField/index.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { ReactNode, FunctionComponent } from "react";
+import styled from "styled-components";
+
+type Orientation = "horizontal" | "vertical";
+
+type LabeledFieldProps = {
+    orientation: Orientation;
+    label: ReactNode;
+    value: ReactNode;
+    margin?: string;
+};
+
+const LabeledField: FunctionComponent<LabeledFieldProps> = ({ orientation, label, value, margin }) => (
+    <Root margin={margin} orientation={orientation}>
+        <Label>{label}</Label>
+        <Value orientation={orientation}>{value}</Value>
+    </Root>
+);
+
+type OrientationProps = {
+    orientation: Orientation;
+};
+
+type RootProps = {
```

Comment:
I feel like extracting props for styled components is not the best idea because:
1. It should be as small as possible. I think if you have many props for a styled comp something might have went wrong along the line.
2. If something is reusable like `OrientationProps` maybe it shouldn't be synchronized with React but just injected once in the `Root` component and managed with styled-components.
3. This should be as important to waste that much screen real estate.

So in this specific example what my perfect world case scenario would look like:
```ts

const Label = styled.span``;

const Value = styled.span``;

const Root = styled.div<{ orientation: Orientation, margin?: string }>`
    display: flex;
    flex-direction:  ${({ orientation }) => orientation === "horizontal" ? "row" : "column"};

    ${({ margin }) => (margin ? `margin: ${margin};` : "")}

    ${Value} {
        margin-left: ${({ orientation }) => orientation === "horizontal" ? "auto" : "0"}
    }
`;
```

Reply:
Also in this specific case you could even drop entire margin-left and orientation and just do
```ts
const Root = styled.div<{ orientation: Orientation, margin?: string }>`
    display: flex;
    flex-direction:  ${({ orientation }) => orientation === "horizontal" ? "row" : "column"};
    justify-content: space-between;
`;
```
I haven't tested it but I think it gives the same result. Then you don't even need `Label` and `Value` and can just use two divs.